### DISCUSSION TITLE
Some questions about data format

### DISCUSSION THREAD

**Discussion by daisysong1** [Project-MONAI/MONAI#6142] [Q&A] [answered]
My original data is CT data, image and label are single-channel 3D data, and they are converted into nii format. 
In my first attempt, the dataloder was successful, and the error occurred in the loss phase. Because the output of the single-channel data input into the UNETR structure is multi-channel 3D data, which is inconsistent with my original label data. So I processed the original label data with one-hot, but the monai structure cannot process the four-dimensional data(axcode require to be three-dimensional). 
I want to know how to adjust such data format.
For some reasons, I haven't downloaded the BTCV dataset yet. Let's discuss here first, hoping to find a solution.

**Comment by KumoLiu** [comment, marked as answer]
Hi @daisysong1, I think you want to one-hot your label from a shape like (1, 96, 96, 96) to (classes, 96, 96, 96). Can't `AsDiscrete` do that? 
https://github.com/Project-MONAI/MONAI/blob/a8302eca3fc31554d4bd30fae84150766ce13472/monai/transforms/post/array.py#L131
You could also do it directly in the loss function like:
```
loss_function = DiceCELoss(to_onehot_y=True, softmax=True)
```
For more details, you could refer to this [tutorial](https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/unetr_btcv_segmentation_3d.ipynb).

Hope it can help you, thanks!

**Reply by daisysong1** [reply]
Thank you very much for your reply.
I just went to download the BTCV dataset, and found that the format of my data is the same with the dataset.This brings me back to my initial problem.
Here is the problem"RuntimeError: cuDNN error: CUDNN_STATUS_NOT_INITIALIZED"
It is the following place that has a problem

> loss.backward()

However, I have checked the version problem.Online evaluation is also that it is not necessarily a version problem. Do you have any suggestions?

**Reply by KumoLiu** [reply]
Hi @daisysong1, there could be many reasons for this error. 
First, you could check your environment, the version of PyTorch and CUDA may mismatch.
Second, you could try to use `print(torch.cuda.is_available())` to check whether your cuda is available.

You could refer to [this installation page](https://docs.monai.io/en/stable/installation.html) for more details about how to prepare your environment.
Hope it can help you, thanks!
